patches/minecraft/net/minecraft/client/gui/recipebook/RecipeTabToggleWidget.java.patch

```diff
@@ -0,0 +1,66 @@
+--- a/net/minecraft/client/gui/recipebook/RecipeTabToggleWidget.java
++++ b/net/minecraft/client/gui/recipebook/RecipeTabToggleWidget.java
+@@ -16,18 +16,19 @@
+ 
+ @OnlyIn(Dist.CLIENT)
+ public class RecipeTabToggleWidget extends ToggleWidget {
+-   private RecipeBookCategories field_193921_u;
++   @Deprecated private RecipeBookCategories field_193921_u;
+    private float field_193922_v;
+ 
++   @Deprecated
+    public RecipeTabToggleWidget(RecipeBookCategories p_i51075_1_) {
+       super(0, 0, 35, 27, false);
+-      this.field_193921_u = p_i51075_1_;
++      this.recipeBookCategory = p_i51075_1_;
+       this.func_191751_a(153, 2, 35, 0, RecipeBookGui.field_191894_a);
+    }
+ 
+    public void func_193918_a(Minecraft p_193918_1_) {
+       ClientRecipeBook clientrecipebook = p_193918_1_.field_71439_g.func_199507_B();
+-      List<RecipeList> list = clientrecipebook.func_202891_a(this.field_193921_u);
++      List<RecipeList> list = clientrecipebook.getRecipes(this.recipeBookCategory);
+       if (p_193918_1_.field_71439_g.field_71070_bA instanceof RecipeBookContainer) {
+          label25:
+          for(RecipeList recipelist : list) {
+@@ -90,7 +91,7 @@
+    }
+ 
+    private void func_193920_a(ItemRenderer p_193920_1_) {
+-      List<ItemStack> list = this.field_193921_u.func_202903_a();
++      List<ItemStack> list = this.recipeBookCategory.getIcon();
+       int i = this.field_191755_p ? -2 : 0;
+       if (list.size() == 1) {
+          p_193920_1_.func_180450_b(list.get(0), this.x + 9 + i, this.y + 5);
+@@ -101,12 +102,13 @@
+ 
+    }
+ 
++   @Deprecated
```

Comment:
Here however a comment would be nice.
You mark this method as Deprecated, however you do not explain anywhere what to do if you want to use the method, aka what the alternatives are, or where the new registry is located.

When you deprecated a field, method or class you should insert a comment what to do in case a modder is using it. This is a higher priority (at least in my eyes) then keeping the patches small.

Reply:
Added comments for deprecated methods